plugins/inputs/prometheus/prometheus.go

```diff
@@ -301,6 +300,15 @@ func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error
 		req.SetBasicAuth(p.Username, p.Password)
 	}
 
+	// Add header parameters
+	for k, v := range p.Headers {
+		if strings.ToLower(k) == "host" {
+			req.Host = v
+		} else {
+			req.Header.Add(k, v)
```

Comment:
Should we check if we will override an previously set header? Like the "Authorization" option above.

Reply:
Well, it would be the case if someone would re-define a header named "Authorization" in the config, otherwise, I don't see the case where it would be overridden, right?

Reply:
You are right.